src/mosquitto_stubs.c

```diff
@@ -21,13 +21,16 @@
 #include <caml/threads.h> 
 #include <caml/callback.h>
 #include <caml/fail.h>
+#include <caml/custom.h>
 #include <caml/unixsupport.h>
 
 #define RESULT_OK caml_alloc(1, 0)
 #define RESULT_ERROR caml_alloc(1, 1)
 
 static value eunix;
 
+static char * callback_id = (char *) 0;
```

Comment:
A counter for generating unique callback IDs. (Using the address of the struct is not failsafe!)